pkg/custom_detectors/regex_varstring.go

```diff
@@ -0,0 +1,44 @@
+package custom_detectors
+
+import (
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// nameGroupRegex matches `{ name . group }` ignoring any whitespace.
+var nameGroupRegex = regexp.MustCompile(`{\s*([a-zA-Z0-9-_]+)\s*(\.\s*[0-9]*)?\s*}`)
+
+// RegexVarString is a string with embedded {name.group} variables. A name may
+// only contain alphanumeric, hyphen, and underscore characters. Group is
+// optional but if provided it must be a non-negative integer. If the group is
+// omitted it defaults to 0.
+type RegexVarString struct {
+	original string
+	// map from name to group
+	variables map[string]int
+}
+
+func NewRegexVarString(original string) RegexVarString {
```

Comment:
__Note__: At some point we might want these to error to ensure there isn't any unexpected behavior, but that addition definitely doesn't need to be a part of this PR, and this is great.

Reply:
Good point. For now we won't be using these function with the focus being on webhooks, but if/when we do implement templates we'll want to be clear with our errors and warnings.